src/main/java/desi/juan/email/api/EmailTools.java

```diff
@@ -0,0 +1,282 @@
+package desi.juan.email.api;
+
+import com.google.common.collect.ImmutableList;
+import com.sun.mail.imap.IMAPFolder;
+import com.sun.mail.pop3.POP3Folder;
+import desi.juan.email.internal.StoredEmail;
+import desi.juan.email.internal.commands.FlagOperations;
+import desi.juan.email.internal.exception.RetrieveEmailException;
+
+import javax.mail.Address;
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.UIDFolder;
+import javax.mail.internet.ContentType;
+import javax.mail.internet.MimeUtility;
+import javax.mail.search.AndTerm;
+import javax.mail.search.ComparisonTerm;
+import javax.mail.search.ReceivedDateTerm;
+import javax.mail.search.SearchTerm;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+
+import static desi.juan.email.api.EmailConstants.DEFAULT_CHARSET;
+import static desi.juan.email.internal.commands.FolderOperations.ALL_MESSAGES;
+import static java.lang.Long.parseLong;
+import static java.lang.String.format;
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.toList;
+import static javax.mail.Flags.Flag.ANSWERED;
+import static javax.mail.Flags.Flag.DELETED;
+import static javax.mail.Flags.Flag.DRAFT;
+import static javax.mail.Flags.Flag.RECENT;
+import static javax.mail.Flags.Flag.SEEN;
+
+public class EmailTools {
```

Comment:
this should part of the api?

Reply:
seems like an internal Utils, but this is going to propagate very quickly, code from this class should be separated by concerns